File: SolutionCodes/Q29.cpp
// 9. Palindrome Number

class Solution {
public:
    bool isPalindrome(int x) {
        if (x<0 || (x!=0 && x%10==0))
            return false;
        int sum = 0;
        while(x>sum)
        {   sum = sum*10 + x%10;
            x/=10;
        }
        return (x==sum||x==sum/10);
    }
};

/*
//sample 0 ms submission

class Solution {
public:
    bool isPalindrome(long int x) {
       long int r,sum=0,t;
        t=x;
        while(t>0)
        {
            r=t%10;
            sum=sum*10+r;
            t/=10;
        }
        if(sum==x)
            return true;
        return false;
    }
};

*/

File: SolutionCodes/Q29.py
# 9. Palindrome Number

class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x<0:
            return False
        new = str(x)
        if x == int(new[::-1]):
            return True
        else:
            return False


# # sample 36 ms submission
#
# class Solution:
#     def isPalindrome(self, x: int) -> bool:
#         if x < 0:
#             return False
#         elif x < 10:
#             return True
#         else:
#             rev = 0
#             dig = x
#             while (x != 0):
#                 rem = x % 10
#                 rev = (rev * 10) + rem
#                 x = x // 10
#             if rev == dig:
#                 return True
#             else:
#                 return False
